src/pages/Invitation.tsx

```diff
@@ -0,0 +1,296 @@
+  const [errors, setErrors] = useState<FORM_ERRORS[]>([])
+  const validationSchema = useMemo(
+    () =>
+      object().shape({
+        password: string()
+          .min(8, FORM_ERRORS.MIN)
+          .matches(RegExp('(.*[a-z].*)'), FORM_ERRORS.LOWERCASE)
+          .matches(RegExp('(.*[A-Z].*)'), FORM_ERRORS.UPPERCASE)
+          .matches(RegExp('(.*\\d.*)'), FORM_ERRORS.NUMBER)
+          .matches(RegExp('[!@#$%^&*(),.?":{}|<>]'), FORM_ERRORS.SPECIAL),
+      }),
+    []
+  )
+  const onInvitation = async () => {
+    const { password } = formFields
+
+    await acceptInvite({
+      variables: {
+        input: {
+          token: token || '',
+          email: email || '',
+          password: password || '',
+        },
+      },
+    })
+  }
+
+  useEffect(() => {
+    validationSchema
+      .validate(formFields, { abortEarly: false })
+      .catch((err) => err)
+      .then((param) => {
+        if (!!param?.errors && param.errors.length > 0) {
+          setErrors(param.errors)
+        } else {
+          setErrors([])
+        }
+      })
+  }, [formFields, validationSchema])
+
+  useShortcuts([
+    {
+      keys: ['Enter'],
+      disabled: errors.length > 0,
+      action: onInvitation,
+    },
+  ])
+
+  return (
+    <Page>
+      <Card>
+        <StyledLogo height={24} />
+        {!!error && !loading ? (
+          <>
+            <Title variant="headline">{translate('text_63246f875e2228ab7b63dcf4')}</Title>
+            <Subtitle $noMargins>{translate('text_63246f875e2228ab7b63dcfe')}</Subtitle>
+          </>
+        ) : !!loading ? (
+          <>
+            <Skeleton variant="text" width={208} height={12} marginBottom={theme.spacing(8)} />
+            <Skeleton variant="text" width={440} height={12} marginBottom={theme.spacing(4)} />
+            <Skeleton variant="text" width={304} height={12} />
+          </>
+        ) : (
+          <>
+            <Title variant="headline">
+              {translate('text_63246f875e2228ab7b63dcd0', {
+                orgnisationName: data?.invite?.organization.name,
+              })}
+            </Title>
+            <Subtitle>{translate('text_63246f875e2228ab7b63dcd4')}</Subtitle>
+
+            {!!acceptInviteError?.graphQLErrors &&
+              acceptInviteError?.graphQLErrors[0] &&
+              acceptInviteError?.graphQLErrors[0]?.extensions?.code ===
+                Lago_Api_Error.UserAlreadyExists && (
+                <ErrorAlert type="danger" data-test="error-alert">
+                  <Typography
+                    color="inherit"
+                    html={translate('text_622f7a3dc32ce100c46a5131', { link: LOGIN_ROUTE })}
+                  />
+                </ErrorAlert>
+              )}
+            <form>
+              <Input
+                disabled
+                name="email"
+                label={translate('text_63246f875e2228ab7b63dcdc')}
+                value={email}
+              />
+
+              <PasswordBlock>
```

Comment:
Looks a lot like the signup... Shouldn't we do one component reusable both for the signup and here as it uses the same logic ?

Reply:
We'll do that in another PR